exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,175 @@
+		l.AdditionalProperties[ddSpanID] = strconv.FormatUint(spanIDToUint64(lr.SpanID()), 10)
+		l.AdditionalProperties[otelSpanID] = lr.SpanID().HexString()
+	}
+	var status string
+
+	// we want to use the serverity that client has set on the log and let datadog backend
+	// decide the appropriate level
+	if lr.SeverityText() != "" {
+		status = lr.SeverityText()
+		l.AdditionalProperties[otelSeverityText] = lr.SeverityText()
+	} else if lr.SeverityNumber() != 0 {
+		status = derviveStatusFromSeverityNumber(lr.SeverityNumber())
+	}
+	l.AdditionalProperties[ddStatus] = status
+	if lr.SeverityNumber() != 0 {
+		l.AdditionalProperties[otelSeverityNumber] = strconv.Itoa(int(lr.SeverityNumber()))
+	}
+	// for datadog to use the same timestamp we need to set the additional property of "@timestamp"
+	if lr.Timestamp() != 0 {
+		// we are retaining the nano second precision in this property
+		l.AdditionalProperties[otelTimestamp] = strconv.FormatInt(lr.Timestamp().AsTime().UnixNano(), 10)
+		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
+	}
+	if sendLogRecordBody {
+		// set the Message
+		l.Message = lr.Body().AsString()
+	}
+
+	var tags = attributes.TagsFromAttributes(res.Attributes())
+	if len(tags) > 0 {
+		tagStr := strings.Join(tags, ",")
+		l.Ddtags = datadog.PtrString(tagStr)
+	}
+	return l
+}
+
+func extractHostNameAndServiceName(resourceAttrs pcommon.Map, logAttrs pcommon.Map) (host string, service string) {
+	if src, ok := attributes.SourceFromAttributes(resourceAttrs, true); ok && src.Kind == source.HostnameKind {
+		host = src.Identifier
+	}
+	// hostName is blank from resource
+	// we need to derive from log attributes
+	if host == "" {
+		if src, ok := attributes.SourceFromAttributes(logAttrs, true); ok && src.Kind == source.HostnameKind {
+			host = src.Identifier
+		}
+	}
+	if s, ok := resourceAttrs.Get(conventions.AttributeServiceName); ok {
+		service = s.AsString()
+	}
+	// serviceName is blank from resource
+	// we need to derive from log attributes
+	if service == "" {
+		if s, ok := logAttrs.Get(conventions.AttributeServiceName); ok {
+			service = s.AsString()
+		}
+	}
+	return host, service
+}
+
+// traceIDToUint64 converts 128bit traceId to 64 bit uint64
+func traceIDToUint64(b [16]byte) uint64 {
+	return binary.BigEndian.Uint64(b[len(b)-8:])
+}
+
+// spanIDToUint64 converts byte array to uint64
+func spanIDToUint64(b [8]byte) uint64 {
+	return binary.BigEndian.Uint64(b[:])
+}
+
+// derviveStatusFromSeverityNumber converts the severity number to log level
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber
+// this is not exactly datadog log levels , but derived from range name from above link
+// see https://docs.datadoghq.com/logs/log_configuration/processors/?tab=ui#log-status-remapper for details on how it maps to datadog level
+func derviveStatusFromSeverityNumber(severity plog.SeverityNumber) string {
```

Comment:
```suggestion
func deriveStatusFromSeverityNumber(severity plog.SeverityNumber) string {
```
I'd propose shortening to `statusFromSeverityNumber`